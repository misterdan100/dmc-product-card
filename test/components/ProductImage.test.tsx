import React from "react";
import renderer from 'react-test-renderer'
import {  ProductCard, ProductImage } from "../../src/components";
import { product1 } from '../data/products'

describe('ProductImage', () => {
    test('must to show default image', () => {
        const wrapper = renderer.create(
            <ProductImage 
                img={'http://'}
            />
        )

        expect(wrapper.toJSON()).toMatchSnapshot()
    })

    test('must to show product card with image component', () => {
        const wrapper = renderer.create(
            <ProductCard product={product1}>
                {() => (
                    <ProductImage />
                )}
            </ProductCard>
        )

        expect(wrapper.toJSON()).toMatchSnapshot()
    })
})