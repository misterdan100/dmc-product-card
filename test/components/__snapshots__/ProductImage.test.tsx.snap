// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductImage must to show default image 1`] = `
<img
  alt="product image"
  className="productImg undefined"
  src="http://"
/>
`;

exports[`ProductImage must to show product card with image component 1`] = `
<div
  className="productCard undefined"
>
  <img
    alt="product image"
    className="productImg undefined"
    src="./coffee-mug.png"
  />
</div>
`;
